src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.ts

```diff
@@ -161,6 +164,14 @@ export class ExerciseDetailsStudentActionsComponent implements OnInit {
                     }
                     if (this.exercise.type === ExerciseType.PROGRAMMING) {
                         this.jhiAlertService.success('artemisApp.exercise.personalRepository');
+                        const tourSetting = this.guidedTourService.guidedTourSettings.filter(setting => setting.guidedTourKey === courseExerciseOverviewTour.settingsKey);
+                        if (tourSetting.length > 0 && tourSetting[0].guidedTourState.toString() === GuidedTourState[GuidedTourState.FINISHED]) {
+                            this.guidedTourService.enableTour(cloneRepositoryTour);
+                            // Set timeout for clone repository button to render
+                            setTimeout(() => {
+                                this.guidedTourService.startTour();
+                            }, 1000);
```

Comment:
How do you define this timeout? Did you also check if this works with slower devices?

Reply:
The timeout duration is self-defined, actually it doesn't take long for the button to render, but I wanted to add some more delay, so that the use can take in the UI changes before the tour pops up again. 

Can we merge the PR for now and I tackle this topic in another PR?

Reply:
Yes, sure. I just wanted to be sure that you have some additional delay.